Update release issue template (#6453)
MaterializeInc/materialize: .github/ISSUE_TEMPLATE/release.md

  - [ ] Ensure that [`doc/user/config.toml`] has the correct version, as updated in the "Final
    Release / create git tag" step

>   - [ ] Ping `@relnotes-team` in `#release` to let them know the release process
>     is finished.
> 
>   - [ ] If this is a major version, ensure that the announcement blog post has
>     been published and announced, if applicable, by pinging the product team in
>     `#release`.

Reviewer: I would take some of the responsibility off of the release manager here:

```suggestion
  - [ ] Ping `@relnotes-team` in `#release` to let them know the release process
    is finished, and that they should publish any appropriate blog posts and release
    announcements
```
